SplitIt/app/src/main/java/project/cs495/splitit/GroupManageActivity.java

```diff
@@ -105,32 +111,32 @@ public void onStop() {
     }
 
     private void deleteGroup() {
-        Group group = (Group) adapter.getItem(currGroupIndex);
-        final DatabaseReference removeGroup = FirebaseDatabase.getInstance().getReference("groups").child(group.getGroupId());
-        if (auth.getCurrentUser().getUid().equals(group.getManagerUID())) {
-            removeGroup.removeValue();
+        Group removedGroup = (Group) adapter.getItem(currGroupIndex);
+        if (auth.getCurrentUser().getUid().equals(removedGroup.getManagerUID())) {
+            database.child("groups").child(removedGroup.getGroupId()).removeValue();
+            database.child("users").child(auth.getCurrentUser().getUid()).child("groups").child(removedGroup.getGroupId()).removeValue();

```

Comment:
We need to make sure to also delete the group reference from all members of the group, not just the owner.